libs/dfi/CMakeLists.txt

```diff
@@ -39,7 +39,7 @@ target_include_directories(dfi PUBLIC
 	$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>
 	$<INSTALL_INTERFACE:include/celix/dfi>
 )
-
+target_compile_options(dfi PRIVATE -O0) #Remove optimalization to ensure libdfi / ffi works
```

Comment:
Correct, I think any optimization when using libdfi can be an issue. 

I also noticed that the Ubuntu nightly builds are failing on the dfi tests. The nightly builds are configured for a RelWithDebInfo build type. 

I agree that this should be an separate issue. I will retest if this the "-O0" can be removed the libdfi, because I expect it is only needed for test code. 

I also expect that the optimization gives issues when combined with using ffi for dynamic proxy functions (dynamically creating a stdcall stack). So I will also try to isolate the problematic tests and create a libdfi ticket with this info.